.github/workflows/test_and_deploy.yml

```diff
@@ -112,11 +168,17 @@ jobs:
     # alternatively, to publish when a GitHub Release is created, use the following rule:
     # if: github.event_name == 'release' && github.event.action == 'published'
     steps:
-      - uses: actions/download-artifact@v3
+      - uses: actions/download-artifact@v4
+        with:
+          # unpacks all the wheels into dist/
+          pattern: wheels-*
+          path: dist
+          merge-multiple: true
+          
+      - uses: actions/download-artifact@v4
```

Comment:
Also need to make sure sdist is bundled, because with v4 it's its own thing and the name doesn't match the pattern `wheel-*`.